### PR TITLE
Create regular.c

### LINKED ISSUE #1093: regular falsi in c
I would like to do that problem. gssoc19

### PR DESCRIPTION
**Pull request by gourabk121**
Fixes #1093

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Regular falsi

#### Languages Used:
- c

#### Files Added:
- regular.c

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Regular_Falsi/regular.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// Girlscript Summer Of Code
```
Comment (jainaman224) [inline Regular_Falsi/regular.c]: Fix file name.

File: Regular_Falsi/regular.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+//that, in modified form, is still in use. In simple terms, the method is the trial and 
+//error technique of using test ("false") values for the variable and then adjusting the test value according
+//to the outcome. This is sometimes also referred to as "guess and check".
+//Versions of the method predate the advent of algebra and the use of equations.
+
+#include<stdio.h>
+#include<math.h>
+
+float
+f(float x) {
```
Comment (jainaman224) [inline Regular_Falsi/regular.c]: Define this is in a single line and use better function name.

File: Regular_Falsi/regular.c [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+int main() {
+    float x1, x2, x3, f1, f2, f3;
+    int i;
+    printf("Enter x1 and x2:");
+    scanf("%f %f", & x1, & x2);
+    do {
+        f1 = f(x1);
+        f2 = f(x2);
+        x3 = x1 - (x2 - x1) / (f2 - f1) * f1;
```
Comment (jainaman224) [inline Regular_Falsi/regular.c]: Add some comments with code at well.

File: Regular_Falsi/regular.c [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        f3 = f(x3);
+        if (f1 * f3 > 0) {
+            x2 = x3;
+        } else {
+            x1 = x3;
+        }
+        i++;
+        printf("Number of iteration %d\t root is %f\n", i, x3);
+    }
+    while (fabs(f3) > 0.001);
```
Comment (jainaman224) [inline Regular_Falsi/regular.c]: What is `fabs`?

**Comment by MastersAbh** [comment]
@gourabk121 , still working on it?

**Comment by gourabk121** [comment]
yeah

File: Regular_Falsi/regular.c (hunk 4 above)
Comment (gourabk121) [reply Regular_Falsi/regular.c]: fabs function works like mod functon. If f3 gets negative then it will make the number positive

**Pull request closed without merging**